app/models/vm_or_template/operations/configuration.rb

```diff
@@ -74,7 +74,7 @@ def disconnect_floppies
   def raw_add_disk(disk_name, disk_size_mb, options = {})
     raise _("VM has no EMS, unable to add disk") unless ext_management_system
     if options[:datastore]
-      datastore = Storage.find_by(:name => options[:datastore])
+      datastore = ext_management_system.storages.find_by(:name => options[:datastore])
```

Comment:
Can we use `host.writable_storages` here?  It could be possible to pick a datastore that exists but the VM's host doesn't have access to.